File: packages/javascript/lib/word/randFood.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * Food and drink, the everyday kind.
 *
 * `randWord({ theme: 'food' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randFood({ language: 'ko', count: 3 }); // ['밥', '떡볶이', '빵']
 * randFood({ language: 'en', count: 3 }); // ['Rice', 'Noodle', 'Dumpling']
 */
export function randFood(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Food words with the language and theme each one came from.
 *
 * @example
 * randFood({ language: 'ko', output: 'detail' });
 * // [{ word: '밥', language: 'ko', theme: 'food' }]
 */
export function randFood(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randFood(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('food', options);
}

File: packages/dart/lib/src/word/rand_food.dart
import 'dart:math';

import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// Food and drink, the everyday kind.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.food] to `randWordDetails` — Dart has
/// no overloads, and twenty-four more functions for it would be twenty-four too many.
///
/// ```dart
/// randFood(language: WordLanguage.ko, count: 3); // [밥, 떡볶이, 빵]
/// randFood(language: WordLanguage.en, count: 3); // [Rice, Noodle, Dumpling]
/// ```
List<String> randFood({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  RandVocabulary vocabulary = RandVocabulary.full,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,

  /// Where the randomness comes from: `Random.secure()` for a value nobody may
  /// predict, `Random(42)` for one that has to come out the same every run.
  Random? random,
}) => randWord(
  language: language,
  theme: WordTheme.food,
  count: count,
  realism: realism,
  vocabulary: vocabulary,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
  random: random,
);
